scripts/addons.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import os
+import re
+from urlparse import urlparse
+from subprocess import call
+from subprocess import check_output
+
+EXTRA_ADDONS_PATH = '/mnt/data/additional_addons/'
+ODOO_CONF = '/opt/odoo/etc/odoo.conf'
+ADDONS_PATH = ['/opt/odoo/sources/odoo/addons']
+SCHEME = 'https://'
+GIT_REPOSITORY_HOSTING_SERVICE = 'github.com'
+DEFAULT_ORGANIZATION = 'OCA'
+
+
+class Repo(object):
+    """
+    oca_dependencies.txt
+
+    For public repo:
+
+    oca-repo -> https://github.com/OCA/oca-repo (branch: master)
+    oca-repo 8.0 -> https://github.com/OCA/oca-repo (branch: 8.0)
+    organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    public-repo_rename https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    public-repo_rename https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+
+    For private repo:
+
+    git@github.com:Elico-Corp/private_repo
+    git@github.com:Elico-Corp/private_repo 8.0
+    private_repo_rename git@github.com:Elico-Corp/private-repo
+    private_repo_rename git@github.com:Elico-Corp/private_repo 8.0
+
+    """
+    def __init__(self, dependent, parent=None):
+        if parent:
+            self.parent = parent
+            self.branch = self.parent.branch
```

Comment:
@seb-elico 

1. This is to initialize one Repo , first the parent.branch can be a default value. 
2. In the self._parse() method , the self.branch will be override to itself's branch if it's specified.
3. In the self._parse() method , the self.branch will not be orerride to None if it's not specified a branch , so it use the parent.branch.

PS. Before the git clone/pull , the self.branch should have a default value from the parent , this was the purpose why I added default value in the  __init__ method.

Above is all the case I think , we can also check self._set_branch() method to make sure what we did change the self.branch.

def _set_branch(self, branch):
        if branch:
            self.branch = branch
        else:
            self.branch = self.parent.branch